egs/wsj/s5/steps/nnet3/components.py

```diff
@@ -347,4 +347,164 @@ def AddClstmLayer(config_lines,
             }
 
 
+def AddBlstmLayer(config_lines,
```

Comment:
I meant something like the equations below.

// This generates a single config corresponding to an LSTM.// based on the
equations in// Sak et. al. "LSTM based RNN architectures for LVCSR", 2014//
We name the components based on the following equations (Eqs 7-15 in paper)//
i(t) = S(Wix * x(t) + Wir * r(t-1) + Wic * c(t-1) + bi)// f(t) = S(Wfx *
x(t) + Wfr * r(t-1) + Wfc * c(t-1) + bf)// c(t) = {f(t) .* c(t-1)} + {i(t)
.* g(Wcx * x(t) + Wcr * r(t-1) + bc)}// o(t) = S(Wox * x(t) + Wor * r(t-1)
+ Woc * c(t) + bo)// m(t) = o(t) .* h(c(t))// r(t) = Wrm * m(t)// p(t) =
Wpm * m(t)// y(t) = Wyr * r(t) + Wyp * p(t) + by// where S : sigmoid//
matrix with feed-forward connections// from the input x(t)// W*x = [Wix^T,
Wfx^T, Wcx^T, Wox^T]^T // matrix with recurrent (feed-back) connections//
from the output projection// W*r = [Wir^T, Wfr^T, Wcr^T, Wor^T]^T // matrix
to generate r(t) and p(t)// m(t)// W*m = [Wrm^T, Wpm^T]^T// matrix to
generate y(t)// Wy* = [Wyr^T, Wyp^T] ™

On Sun, Oct 25, 2015 at 2:06 PM, Yiming Wang <notifications@github.com>
wrote:

> In egs/wsj/s5/steps/nnet3/components.py
> <https://github.com/freewym/kaldi/pull/1#discussion_r42947914>:
>
> > @@ -347,4 +347,164 @@ def AddClstmLayer(config_lines,
> >              }
> >
> >
> > +def AddBlstmLayer(config_lines,
>
> I have added equations as comments above every component-node.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/freewym/kaldi/pull/1/files#r42947914>.
>